sdk/ml/azure-ai-ml/azure/ai/ml/operations/_capability_hosts_operations.py

```diff
@@ -0,0 +1,345 @@
+                if capability_host.storage_connections is None or len(capability_host.storage_connections) == 0:
+                    capability_host.storage_connections = self.__get_default_storage_connections()
+
+            capability_host_resource = (
+                capability_host._to_rest_object_for_hub()
+                if workspace._kind == WorkspaceKind.HUB
+                else capability_host._to_rest_object_for_project()
+            )
+
+            # pylint: disable=unused-argument, docstring-missing-param
+            def callback(_: Any, deserialized: Any, args: Any) -> CapabilityHost:
+                """Callback to be called after completion of begin_create_or_update operation
+
+                :return: Deserialized CapabilityHost.
+                :rtype: ~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost
+                """
+
+                return CapabilityHost._from_rest_object(deserialized)
+
+            poller = self._capability_hosts_operations.begin_create_or_update(
+                resource_group_name=self._resource_group_name,
+                workspace_name=self._workspace_name,
+                name=capability_host.name,
+                body=capability_host_resource,
+                polling=True,
+                **kwargs,
+                cls=callback,
+            )
+            return poller
+
+        except Exception as ex:
+            if isinstance(ex, (ValidationException, SchemaValidationError)):
+                log_and_raise_error(ex)
+            raise ex
+
+    @experimental
+    @distributed_trace
+    @monitor_with_activity(ops_logger, "CapabilityHost.Delete", ActivityType.PUBLICAPI)
+    def begin_delete(
+        self,
+        name: str,
+        **kwargs: Any,
+    ) -> LROPoller[None]:
+        """Delete capability host.
+
+        :param name: capability host name.
+        :type name: str
+        :return: A poller for deletion status
+        :rtype: ~azure.core.polling.LROPoller[None]
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/ml_samples_capability_host.py
+                :start-after: [START capability_host_delete_operation]
+                :end-before: [END capability_host_delete_operation]
+                :language: python
+                :dedent: 8
+                :caption: Delete example.
+        """
+        poller = self._capability_hosts_operations.begin_delete(
+            resource_group_name=self._resource_group_name,
+            workspace_name=self._workspace_name,
+            name=name,
+            polling=True,
+            **kwargs,
+        )
+        return poller
+
+    def __get_default_storage_connections(self) -> List[str]:
+        """Retrieve the default storage connections for a capability host.
+
+        :return: A list of default storage connections.
+        :rtype: List[str]
+        """
+        return [f"{self._workspace_name}/{DEFAULT_STORAGE_CONNECTION_NAME}"]
```

Comment:
can we have a work item to push defaults like this into the service? This can end up being very brittle if that default name ever changes.

Reply:
agree and will have one.